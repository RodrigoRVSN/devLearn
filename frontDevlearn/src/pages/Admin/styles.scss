#home-page {
  display: flex;
  flex-direction: column;
  padding: 2rem 5rem;
  gap: 2rem;
  align-items: center;
  text-align: center;

  button {
    width: 10rem;
  }

  @media (max-width: 550px) {
    padding: 2rem 0.5rem;
  }

  h4 {
    margin-bottom: 2rem;
  }

  .module-card {

    cursor: pointer;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 2rem;
    border-radius: 1rem;
    padding: 2rem;
    background-color: var(--medium-primary);
    color: var(--background);
    border: 5px solid var(--dark-primary);
    &:hover {
      border: 5px solid var(--dark-secondary);
      background-color: var(--background);
      color: var(--light-secondary);
    }
    h2 {
      font-size: 3rem;
    }
  }
}
