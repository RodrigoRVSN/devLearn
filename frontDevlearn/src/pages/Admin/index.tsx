import { Grid } from "@material-ui/core";
import { api } from "../../services/api";

import { useEffect, useState } from "react";
import { IClasses } from "../../@types/IClasses";
import { useModules } from "../../hooks/useModules";

import { AddModule } from "../../components/AddModule";
import { AddClasses } from "../../components/AddClasses";
import { Button } from "../../components/Button";
import EditModule from "../../components/EditModule";
import DeleteModule from "../../components/DeleteModule";
import ClassesAdminMap from "../../components/ClassesAdminMap";
import Loading from "../../components/Loading";
import HomeIcon from "@material-ui/icons/Home";
import "./styles.scss";
import { useHistory } from "react-router-dom";

export function Admin(): JSX.Element {
  const [classes, setClasses] = useState<IClasses[]>();
  const { modules, loading } = useModules();
  const history = useHistory();

  useEffect(() => {
    const token = localStorage.getItem("token");
    if (!token) {
      history.push("/");
    }
  }, [history, modules]);

  /* Pega as aulas do módulo clicado */

  function classesInModule(module_id: string[]) {
    async function getClasses() {
      await api.get(`/${module_id}/classes`).then((res) => {
        res.data.sort(function (a: IClasses, b: IClasses) {
          if (a.name_class < b.name_class) return -1;
          if (a.name_class > b.name_class) return 1;
          return 0;
        });
        setClasses(res.data);
      });
    }
    getClasses();
  }

  /* Copia o ID ao clicar no botão COPIAR ID */

  function copyId(module_id: string[]) {
    navigator.clipboard.writeText(module_id.toString());
  }

  /* Renderização da página */

  return (
    <>
      <div id="home-page">
        <Button onClick={() => history.push("/")}>
          <HomeIcon />
        </Button>
        <h2>PÁGINA DE ADMINISTRAÇÃO</h2>
        <h4>
          Obs: Basta clicar no botão "COPIAR ID" para o ID do respectivo módulo
          ir automaticamente para a sua área de transferência.
        </h4>
        <AddModule />
        <AddClasses />
        {loading ? (
          <>
            <Grid
              spacing={4}
              container
              style={{
                alignItems: "center",
                justifyContent: "center",
                textAlign: "center",
              }}
            >
              {modules?.map((item, index) => {
                return (
                  <Grid key={index} item xs={11} md={4} lg={3}>
                    <div
                      className="module-card toLeft"
                      onClick={() => classesInModule(item.id)}
                    >
                      <h2>{item.module}</h2>
                      <Button onClick={() => copyId(item.id)}>COPIAR ID</Button>
                      <h3>
                        {item.amount === 0
                          ? "Não há aulas aqui"
                          : item.amount === 1
                          ? "1 aula"
                          : `${item.amount} aulas`}
                      </h3>
                      <div className="button-container">
                        <DeleteModule module_id={item.id} />
                        <EditModule module_id={item.id} />
                      </div>
                    </div>
                  </Grid>
                );
              })}
            </Grid>
          </>
        ) : (
          <Loading />
        )}
        <span id="spacer">AULAS</span>
        {classes && <ClassesAdminMap classes={classes} />}
      </div>
    </>
  );
}
