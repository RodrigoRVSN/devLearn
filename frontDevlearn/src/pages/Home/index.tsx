import { Grid } from "@material-ui/core";
import { useState } from "react";

import { api } from "../../services/api";
import { IClasses } from "../../@types/IClasses";
import { useModules } from "../../hooks/useModules";

import Header from "../../components/Header";
import ClassesMap from "../../components/ClassesMap";
import "./styles.scss";
import Loading from "../../components/Loading";

export function Home(): JSX.Element {
  const { modules, loading } = useModules();
  const [classes, setClasses] = useState<IClasses[]>();

  /* Pega as aulas do módulo clicado */

  function classesInModule(module_id: string[]) {
    navigator.clipboard.writeText(module_id.toString());
    async function getClasses() {
      await api.get(`/${module_id}/classes`).then((res) => {
        res.data.sort(function (a: IClasses, b: IClasses) {
          if (a.name_class < b.name_class) return -1;
          if (a.name_class > b.name_class) return 1;
          return 0;
        });
        setClasses(res.data);
      });
    }
    getClasses();
  }

  /* Renderização da página */

  return (
    <>
      <Header />
      <div id="home-page">
        {loading ? (
          <>
            <h2>MÓDULOS</h2>
            <h4>
              Selecione os módulos! Ao abrir um módulo, as aulas estarão no fim
              da página.
            </h4>
            {modules?.length === 0 && <h3>Não há módulos no momento</h3>}
            <Grid
              spacing={4}
              container
              style={{
                alignItems: "center",
                justifyContent: "center",
                textAlign: "center",
              }}
            >
              {modules?.map((item, index) => {
                return (
                  <Grid key={index} item xs={11} md={4} lg={3}>
                    <div
                      className="module-card toLeft"
                      onClick={() => classesInModule(item.id)}
                    >
                      <h2>{item.module}</h2>
                      <h3>
                        {item.amount === 1
                          ? "1 aula"
                          : item.amount > 1
                          ? `${item.amount} aulas`
                          : `Sem aulas`}
                      </h3>
                    </div>
                  </Grid>
                );
              })}
            </Grid>
          </>
        ) : (
          <Loading />
        )}
        <span id="spacer">AULAS</span>
        {classes && <ClassesMap classes={classes} />}
      </div>
    </>
  );
}
