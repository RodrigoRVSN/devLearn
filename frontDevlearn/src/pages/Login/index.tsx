import { api } from "../../services/api";
import { Paper, TextField } from "@material-ui/core";
import { Button } from "../../components/Button";
import { useForm } from "react-hook-form";

import "./styles.scss";
import { useHistory } from "react-router-dom";
import { useState } from "react";

export function Login(): JSX.Element {
  const { handleSubmit, register } = useForm();
  const history = useHistory();
  const [error, setError] = useState("");

  const onSubmit = (submitted: any) => {
    function getItems() {
      api
        .post(`/login`, {
          email: submitted.email,
          password: submitted.password,
        })
        .then((res) => {
          localStorage.setItem("token", res.data);
          history.push("/admin");
        })
        .catch((err) => {
          setError("Login está incorreto!");
        });
    }
    getItems();
  };

  return (
    <>
      <div id="register-page">
        <Paper elevation={10} className="box-container">
          <h2>DEV LEARN</h2>
          <h1>Faça seu login!</h1>

          <form onSubmit={handleSubmit(onSubmit)}>
            <TextField
              required
              {...register("email")}
              label="Email"
              type="email"
              variant="outlined"
              placeholder="rodrigovictor81@gmail.com"
            />
            <TextField
              required
              {...register("password")}
              label="Senha"
              type="password"
              variant="outlined"
              placeholder="*******"
            />
            <p>{error}</p>
            <Button type="submit">ENTRAR</Button>
          </form>

          <h3>
            Não tem conta? <a href="/register">Cadastre-se</a>
          </h3>
        </Paper>
      </div>
    </>
  );
}
