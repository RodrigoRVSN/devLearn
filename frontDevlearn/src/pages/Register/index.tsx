import { api } from "../../services/api";
import { Paper, TextField } from "@material-ui/core";
import { Button } from "../../components/Button";
import { useForm } from "react-hook-form";

import "./styles.scss";
import { useHistory } from "react-router-dom";

export function Register(): JSX.Element {
  const { handleSubmit, register } = useForm();
  const history = useHistory();

  const onSubmit = (submitted: any) => {
    function getItems() {
      api.post(`/users`, {
        email: submitted.email,
        password: submitted.password,
      });
    }
    getItems();
    history.push("/login");
  };

  return (
    <>
      <div id="register-page">
        <Paper elevation={10} className="box-container toLeft">
          <h2>DEV LEARN</h2>
          <h1>Faça seu cadastro!</h1>

          <form onSubmit={handleSubmit(onSubmit)}>
            <TextField
              required
              {...register("email")}
              label="Email"
              type="email"
              variant="outlined"
              placeholder="rodrigovictor81@gmail.com"
            />
            <TextField
              required
              {...register("password")}
              label="Senha"
              type="password"
              variant="outlined"
              placeholder="*******"
            />
            <Button type="submit">CADASTRAR</Button>
          </form>

          <h3>
            Já tem conta? <a href="/login">Faça login</a>
          </h3>
        </Paper>
      </div>
    </>
  );
}
