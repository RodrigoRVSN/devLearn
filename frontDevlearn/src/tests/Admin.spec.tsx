import React from 'react'
import { render } from '@testing-library/react'
import { Admin } from '../pages/Admin'

describe('<Admin />', () => {
  it('should render the admin page', () => {
    const { container } = render(<Admin />)
    expect(container).toMatchSnapshot()
  })
})
