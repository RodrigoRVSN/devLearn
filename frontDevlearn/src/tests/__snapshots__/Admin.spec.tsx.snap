// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> should render the admin page 1`] = `
<div>
  <div
    id="home-page"
  >
    <button
      class="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </button>
    <h2>
      PÁGINA DE ADMINISTRAÇÃO
    </h2>
    <h4>
      Obs: Basta clicar no botão COPIAR ID para o ID do respectivo módulo ir automaticamente para a sua área de transferência. Atualize a página para ver o estado do banco de dados em tempo real
    </h4>
    <button
      class="button"
    >
      Adicionar módulo
    </button>
    <button
      class="button"
    >
      Adicionar aula
    </button>
    <div
      class="loading-page"
    >
      <div
        class="loading"
      >
        <p>
          DEV LEARN
        </p>
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
    <span
      id="spacer"
    >
      AULAS
    </span>
  </div>
</div>
`;
